[Chore] Create CI env in Jenkins for bulk Expression Atlas in GCP k8s cluster

ke4 opened the pull request:
Changes:
- Add yaml configuration files for:
  - for `gxa-data`
  - for `gxa-solrcloud`

ke4 committed:
Add gxa-data folder

ke4 committed:
Add gxa-solrcloud folder

ke4 committed:
Fix storage size for gxa-data-rox

ke4 committed:
Fix storage size for gxa solr cloud

ke4 committed:
Fix experiment ids and folder location

ke4 committed:
Fix storage size for gxa-data-rwo

alfonsomunozpomer committed:
YAML files to create a Postgres DB migrated to the latest schema version

(cherry picked from commit 4bb2f64f243b58d78b2902ec5e44bcb49befdcf5)

alfonsomunozpomer committed:
Job to populate Postgres

alfonsomunozpomer committed:
Remove kustomization, we‘ll generate secrets the old-fashioned way

alfonsomunozpomer committed:
SolrCloud manifest four bulk using the Solr Operator

alfonsomunozpomer committed:
Volume and jobs to create the bioentities collection in bulk

alfonsomunozpomer committed:
Docker image with CLI and repos to populate Postgres and SolrCloud

alfonsomunozpomer committed:
Fix typo in repo URL (!)

alfonsomunozpomer committed:
Set requests and limits to the same memory amount to make it effective

alfonsomunozpomer committed:
Job to write the experiment JSONL files for the bulk-analytics collection

alfonsomunozpomer committed:
Restore to more conservative memory limits

alfonsomunozpomer committed:
Fix container name

alfonsomunozpomer committed:
Add bulk-analytics loader job

alfonsomunozpomer committed:
Remove invalid mounts

alfonsomunozpomer committed:
Increase size of ephemeral resources

alfonsomunozpomer committed:
Add ephemeral volume to write the chunks of the JSONL files

alfonsomunozpomer committed:
Bulk experiment design volumes

alfonsomunozpomer committed:
Add exit statement to LFTP script

alfonsomunozpomer committed:
Set volume size to 1Gi (or GKE will complain)

alfonsomunozpomer committed:
Add ontology files for bulk Expression Atlas

alfonsomunozpomer committed:
Use mirror to copy a directory and its contents (!)

alfonsomunozpomer committed:
Mirror doesn’t use -o

alfonsomunozpomer committed:
Add initial section to README (populated ROX volumes)

alfonsomunozpomer committed:
Add first section to README (bioentity properties volume)

alfonsomunozpomer committed:
Add second section to README (Gradle RO dependency cache volume)

alfonsomunozpomer committed:
Restructure first section Before We Start and include instructions to inspect volume contents

alfonsomunozpomer committed:
Initial refactoring of bulk data volumes

alfonsomunozpomer committed:
Add section for bulk Expression Atlas data volumes

alfonsomunozpomer committed:
Merge Postgres deployment, service and migrator job into a single YAML file

alfonsomunozpomer committed:
Write section in README for Postgres

alfonsomunozpomer committed:
Reduce SolrCloud cluster to two nodes and draft README section

alfonsomunozpomer committed:
Increase ephemeral storage and reduce commit docs to half the default value of 1M

alfonsomunozpomer committed:
Add region/zone settings to YAML and a section in the README on the rationale for this

alfonsomunozpomer committed:
Set CPU resource requests and limits

alfonsomunozpomer committed:
Last touches of the README doc

alfonsomunozpomer committed:
Close double quotes in env variable value (!)

alfonsomunozpomer committed:
Scale Solr back to 4 replicas, it performs much better

alfonsomunozpomer committed:
Use read-only volume for experiment designs

alfonsomunozpomer committed:
Give the JSONL jobs a bit more memory

alfonsomunozpomer committed:
Add section in README that explains how to forward Solr web UI for inspection

alfonsomunozpomer committed:
Fix typo in Quay instructions

alfonsomunozpomer committed:
Adjust timeout of bioentities JSONL job

alfonsomunozpomer committed:
Fix section headers

alfonsomunozpomer committed:
Add missing experiments from fixtures and adjust size of volumes

alfonsomunozpomer committed:
Exclude old versions of the test experiments to save storage

alfonsomunozpomer committed:
Enforce use of the same zone for RWO volumes

ke4 committed:
Merge branch 'main' into chore/add_ci_env_for_bulk

amnonkhen commented:
I would suggest to make this into a helm chart.

amnonkhen commented on gxa-postgres/gxa-postgres-populator.yaml:
```diff
@@ -0,0 +1,71 @@
+            - name: POSTGRES_USER
+              value: "atlasdev"
+            - name: POSTGRES_PASSWORD
+              value: "atlasdev"
```
It's unsafe to keep password here, isn't it?

ke4 committed:
Removed E-MTAB-2770, E-MTAB-5200, E-MTAB-5423 from data sets to synch with local env

Removed "E-MTAB-2770", "E-MTAB-5200" and "E-MTAB-5423" from the EXP_IDS environment variable. This ensures the configuration aligns with our local environment.

ke4 committed:
Removed E-MTAB-2770, E-MTAB-5200, E-MTAB-5423 from 2 another yaml files

Removed "E-MTAB-2770", "E-MTAB-5200" and "E-MTAB-5423" from the EXP_IDS environment variable. This ensures the configuration aligns with our local environment.

amnonkhen committed:
remove E-MTAB-3579 E-PROT-15 which are not in local env and postgres populator

amnonkhen committed:
fix broken artifactoy url #285

amnonkhen committed:
fix experiment loading commands ebi-gene-expression-group/atlas-web-bulk#285

ke4 committed:
Update YAML files for storage and image changes with git setup

Increased storage requests to 200Gi and replaced the image with `openjdk:11` for both JSONL and populator jobs. Added commands to install `git` and clone required repositories for proper script execution. Adjusted paths accordingly for loading and processing tasks.

ke4 committed:
Add a bash startup command to the container

Added a shell command to the container configuration to enhance flexibility in executing tasks. This change ensures better control and customization of runtime behavior.

ke4 committed:
Update bioentities populator to use OpenJDK image

Replaced the Docker image with OpenJDK 11 and added commands for dynamically cloning and setting up the required repository. Introduced a zone-based `nodeSelector` for better resource targeting. This improves flexibility and ensures reproducibility during deployment.
